# Base original

In [1]:
import pandas as pd
df = pd.read_csv('C:\\Users\\camila.gusmao\\Downloads\\dados_TB.csv')
df.head(1)

ID DT_PEDIDO DT_COLETA  ATENDIMENTO  CD_ORI_ATEND   ORIGEM  PRONTUARIO  \
0   0  01/02/16  01/02/16      1803663            13  EXTERNO           0   

   CD_SETOR                       SETOR  CD_CONVENIO           CONVENIO  \
0       219  SAM-Serviço de Ambulatório            2  SUS - AMBULATORIO   

   CD_EXA_LAB                                          EXAME MATERIAL  \
0         140  PESQUISA DE BAAR (BACILOSCOPIA) - DIAGNÓSTICA  ESCARRO   

   AMOSTRA OBSERVACAO                                          RESULTADO  
0  1022027        NaN  PESQUISA DE BAAR NEGATIVA EM 100 CAMPOS EXAMIN...

Verificando os tipos de materiais

In [2]:
print('Quantidade de materiais: ', df.MATERIAL.value_counts().size)
df.MATERIAL.value_counts()

Quantidade de materiais:  51


LAVADO BRONQUICO                  3796
ESCARRO                           3101
LIQUIDO PLEURAL                   1538
LAVADO BRONCOALVEOLAR              868
LIQUOR (PUNCAO LOMBAR)             409
BIOPSIA                            304
DIVERSOS                           215
ASPIRADO TRAQUEAL                  197
LINFONODO                          121
ESCARRO 2ª AMOSTRA                  91
LIQUIDO ASCITICO                    85
FRAGMENTO PULMONAR                  76
URINA                               67
ESCARRO INDUZIDO (BAAR)             66
ESCARRO 1ª AMOSTRA                  65
LAVADO GASTRICO                     64
LINFA                               59
ESCARRO 3ª AMOSTRA                  50
ABSCESSO                            49
LIQUIDO PERICARDICO                 49
FRAGMENTO DE TECIDO                 48
LIQUIDO SINOVIAL                    42
FRAGMENTO DE PLEURA                 41
ASPIRADO MEDULA OSSEA               40
FRAGMENTO OSSEO                     31
BIOPSIA PULMONAR A CEU AB

# Tratamento dos dados

Renomeando material com erro de ortografia

In [3]:
df["MATERIAL"].replace({"SWAB DE SECREÇÃO DE LESÃO ÓSSE": "SWAB DE SECREÇÃO DE LESÃO ÓSSEA"}, inplace=True)

Segue a lista de materiais e suas respectivas quantidades que serão removidos do dataset:

- DIVERSOS                       215
- ASPIRADO TRAQUEAL              197
- LAVADO GASTRICO                 64
- ABSCESSO                        49
- FRAGMENTO DE TECIDO             48
- ASPIRADO TRANSTRAQUEAL          12
- BIOPSIA TECIDOS PROFUNDOS       11
- SECREÇÃO DE ORGÃOS GENITAIS      4
- BIOPSIA TRANS-BRONQUICA          1
- EMPIEMA EXTRA-DURAL              1

In [4]:
df.drop(df[df.MATERIAL == 'DIVERSOS'].index, inplace=True)
df.drop(df[df.MATERIAL == 'ASPIRADO TRAQUEAL'].index, inplace=True)
df.drop(df[df.MATERIAL == 'LAVADO GASTRICO'].index, inplace=True)
df.drop(df[df.MATERIAL == 'ABSCESSO'].index, inplace=True)
df.drop(df[df.MATERIAL == 'FRAGMENTO DE TECIDO'].index, inplace=True)
df.drop(df[df.MATERIAL == 'ASPIRADO TRANSTRAQUEAL'].index, inplace=True)
df.drop(df[df.MATERIAL == 'BIOPSIA TECIDOS PROFUNDOS'].index, inplace=True)
df.drop(df[df.MATERIAL == 'SECREÇÃO DE ORGÃOS GENITAIS'].index, inplace=True)
df.drop(df[df.MATERIAL == 'BIOPSIA TRANS-BRONQUICA'].index, inplace=True)
df.drop(df[df.MATERIAL == 'EMPIEMA EXTRA-DURAL'].index, inplace=True)


# Criação dos grupos

In [5]:
tb_df = df.copy()
tb_df = df.assign(GRUPO = '')

for index, row in tb_df.iterrows():
    material = row['MATERIAL']
    mat_pulmonar = ['LAVADO BRONQUICO', 'LAVADO BRONCOALVEOLAR', 'ASPIRADO BRONQUICO', 'ESCOVADO BRÔNQUICO', 'ASPIRADO TRAQUEAL', 'ASPIRADO TRANSTRAQUEAL', 'ESCARRO', 'ESCARRO 1ª AMOSTRA', 'ESCARRO 2ª AMOSTRA', 'ESCARRO 3ª AMOSTRA', 'ESCARRO INDUZIDO (BAAR)', 'BIOPSIA', 'BIOPSIA PULMONAR A CEU ABERTO', 'BIOPSIA PULMONAR TRANS-TORACIC', 'NODULO PULMONAR', 'FRAGMENTO PULMONAR']
    # mat_pleural = ['LIQUIDO PLEURAL (POSIT RUIM)', 'EMPIEMA PLEURAL++', 'FRAGMENTO DE PLEURA++']
    mat_pleural = ['LIQUIDO PLEURAL', 'EMPIEMA PLEURAL', 'FRAGMENTO DE PLEURA']
    mat_ganglionar = ['LINFONODO', 'LINFA', 'ASPIRADO GANGLIONAR']
    mat_meningea = ['LIQUOR (PUNCAO LOMBAR)', 'LIQUOR (DVE)', 'LIQUOR (DVP)']
    mat_outros = ['SANGUE', 'ASPIRADO MEDULA OSSEA', 'SECREÇÃO CUTÂNEA', 'SECRECAO DE FERIDA', 'BIOPSIA DE PELE', 'BIOPSIA SUB-CUTANEA', 'LIQUIDO PERICARDICO', 'BIOPSIA DE OSSOS', 'FRAGMENTO OSSEO', 'SECREÇÃO DE LESÃO ÓSSEA', 'SWAB DE SECREÇÃO DE LESÃO ÓSSEA', 'LIQUIDO ASCITICO', 'LIQUIDO ABDOMINAL', 'LIQUIDO PERITONEAL', 'MÚSCULOESQUELÉTICO (MSK)', 'LIQUIDO SINOVIAL', 'FRAGMENTO SINOVIAL', 'LIQUIDO ARTICULAR', 'DIVERSOS', 'EMPIEMA EXTRA-DURAL', 'FRAGMENTO DE TECIDO', 'SECREÇÃO DE ORGÃOS GENITAIS', 'BIOPSIA TECIDOS PROFUNDOS', 'URINA', 'ABSCESSO', 'LAVADO GASTRICO (PED)', 'BIOPSIA TRANS-BRONQUICA']
    
    if material in mat_pulmonar:
        tb_df.loc[index, 'GRUPO'] = 'PULMONAR'
    if material in mat_pleural:
        tb_df.loc[index, 'GRUPO'] = 'PLEURAL'
    if material in mat_ganglionar:
        tb_df.loc[index, 'GRUPO'] = 'GANGLIONAR'
    if material in mat_meningea:
        tb_df.loc[index, 'GRUPO'] = 'MENINGEA'
    if material in mat_outros:
        tb_df.loc[index, 'GRUPO'] = 'OUTROS'

tb_df.to_csv('C:\\Users\\camila.gusmao\\Downloads\\dados_TB-com-grupo.csv')

In [6]:
tb_grupo_df = pd.read_csv('C:\\Users\\camila.gusmao\\Downloads\\dados_TB-com-grupo.csv')
tb_grupo_df.head(1)

Unnamed: 0  ID DT_PEDIDO DT_COLETA  ATENDIMENTO  CD_ORI_ATEND   ORIGEM  \
0           0   0  01/02/16  01/02/16      1803663            13  EXTERNO   

   PRONTUARIO  CD_SETOR                       SETOR  CD_CONVENIO  \
0           0       219  SAM-Serviço de Ambulatório            2   

            CONVENIO  CD_EXA_LAB  \
0  SUS - AMBULATORIO         140   

                                           EXAME MATERIAL  AMOSTRA OBSERVACAO  \
0  PESQUISA DE BAAR (BACILOSCOPIA) - DIAGNÓSTICA  ESCARRO  1022027        NaN   

                                           RESULTADO     GRUPO  
0  PESQUISA DE BAAR NEGATIVA EM 100 CAMPOS EXAMIN...  PULMONAR

Quantidade de atendimentos por grupo

In [7]:
tb_grupo_df.GRUPO.value_counts()

PULMONAR      8494
PLEURAL       1601
MENINGEA       443
OUTROS         382
GANGLIONAR     196
Name: GRUPO, dtype: int64

# Criação das categorias

In [8]:
tb_cat = tb_grupo_df.copy()
tb_cat = tb_cat.assign(CATEGORIA = '')

In [9]:
broncoscopia = ['LAVADO BRONQUICO', 'LAVADO BRONCOALVEOLAR', 'ASPIRADO BRONQUICO', 'ESCOVADO BRÔNQUICO']
escarro = ['ESCARRO', 'ESCARRO 1ª AMOSTRA', 'ESCARRO 2ª AMOSTRA', 'ESCARRO 3ª AMOSTRA']
escarro_induzido = ['ESCARRO INDUZIDO (BAAR)']
biopsia = ['BIOPSIA', 'BIOPSIA PULMONAR A CEU ABERTO', 'BIOPSIA PULMONAR TRANS-TORACIC', 'NODULO PULMONAR', 'FRAGMENTO PULMONAR']
liq_pleural = ['LIQUIDO PLEURAL']
emp_pleural = ['EMPIEMA PLEURAL']
frag_pleura = ['FRAGMENTO DE PLEURA']
linf = ['LINFONODO', 'LINFA', 'ASPIRADO GANGLIONAR']
liquor = ['LIQUOR (PUNCAO LOMBAR)', 'LIQUOR (DVE)', 'LIQUOR (DVP)']
acom_hemat = ['SANGUE', 'ASPIRADO MEDULA OSSEA']
acom_osseo = ['BIOPSIA DE OSSOS', 'FRAGMENTO OSSEO', 'SECREÇÃO DE LESÃO ÓSSEA', 'SWAB DE SECREÇÃO DE LESÃO ÓSSEA']
liq_per = ['LIQUIDO PERICARDICO']
liq_abd = ['LIQUIDO ASCITICO', 'LIQUIDO ABDOMINAL', 'LIQUIDO PERITONEAL']
acom_musc_esq = ['LIQUIDO SINOVIAL', 'FRAGMENTO SINOVIAL', 'LIQUIDO ARTICULAR']
acom_pele = ['SECREÇÃO CUTÂNEA', 'SECRECAO DE FERIDA', 'BIOPSIA DE PELE', 'BIOPSIA SUB-CUTANEA']
urina = ['URINA']

# mat_pulmonar = ['ASPIRADO TRAQUEAL', 'ASPIRADO TRANSTRAQUEAL']
# mat_outros = ['MÚSCULOESQUELÉTICO (MSK)', 'DIVERSOS', 'EMPIEMA EXTRA-DURAL', 'FRAGMENTO DE TECIDO', 'SECREÇÃO DE ORGÃOS GENITAIS', 'BIOPSIA TECIDOS PROFUNDOS', , 'ABSCESSO', 'LAVADO GASTRICO (PED)', 'BIOPSIA TRANS-BRONQUICA']

for index, row in tb_cat.iterrows():
    material = row['MATERIAL']

    if material in broncoscopia:
        tb_cat.loc[index, 'CATEGORIA'] = 'BRONCOSCOPIA'
    if material in escarro:
        tb_cat.loc[index, 'CATEGORIA'] = 'ESCARRO'
    if material in escarro_induzido:
        tb_cat.loc[index, 'CATEGORIA'] = 'ESCARRO INDUZIDO (BAAR)'
    if material in biopsia:
        tb_cat.loc[index, 'CATEGORIA'] = 'BIÓPSIA'
    if material in liq_pleural:
        tb_cat.loc[index, 'CATEGORIA'] = 'LÍQUIDO PLEURAL'
    if material in emp_pleural:
        tb_cat.loc[index, 'CATEGORIA'] = 'EMPIEMA PLEURAL'
    if material in frag_pleura:
        tb_cat.loc[index, 'CATEGORIA'] = 'FRAGMENTO DE PLEURA'
    if material in linf:
        tb_cat.loc[index, 'CATEGORIA'] = 'ACOMETIMENTO LINFÁTICO'
    if material in liquor:
        tb_cat.loc[index, 'CATEGORIA'] = 'LIQUOR'
    if material in acom_hemat:
        tb_cat.loc[index, 'CATEGORIA'] = 'ACOMETIMENTO HEMATOLÓGICO'
    if material in acom_osseo:
        tb_cat.loc[index, 'CATEGORIA'] = 'ACOMETIMENTO ÓSSEO'
    if material in liq_per:
        tb_cat.loc[index, 'CATEGORIA'] = 'LIQUIDO PERICÁRDICO'
    if material in liq_abd:
        tb_cat.loc[index, 'CATEGORIA'] = 'LÍQUIDO EM CAVIDADE ABDOMINAL'
    if material in acom_musc_esq:
        tb_cat.loc[index, 'CATEGORIA'] = 'ACOMETIMENTO MÚSCULOESQUELÉTICO'
    if material in acom_pele:
        tb_cat.loc[index, 'CATEGORIA'] = 'ACOMETIMENTO DE PELE E SUBCUTÂNEO'
    if material in urina:
        tb_cat.loc[index, 'CATEGORIA'] = 'URINA'    

tb_cat["MATERIAL"].replace({"ESCARRO": "NÃO ESPECIFICADO"}, inplace=True)
tb_cat["MATERIAL"].replace({"BIOPSIA": "NÃO ESPECIFICADA"}, inplace=True)

tb_cat.to_csv('C:\\Users\\camila.gusmao\\Downloads\\dados_TB-com-grupo-e-categorias.csv')

In [10]:
import pandas as pd

In [11]:
tb_grupo_cat_df = pd.read_csv('C:\\Users\\camila.gusmao\\Downloads\\dados_TB-com-grupo-e-categorias.csv')
tb_grupo_cat_df.head(1)

Unnamed: 0  Unnamed: 0.1  ID DT_PEDIDO DT_COLETA  ATENDIMENTO  \
0           0             0   0  01/02/16  01/02/16      1803663   

   CD_ORI_ATEND   ORIGEM  PRONTUARIO  CD_SETOR  ... CD_CONVENIO  \
0            13  EXTERNO           0       219  ...           2   

            CONVENIO CD_EXA_LAB  \
0  SUS - AMBULATORIO        140   

                                           EXAME          MATERIAL  AMOSTRA  \
0  PESQUISA DE BAAR (BACILOSCOPIA) - DIAGNÓSTICA  NÃO ESPECIFICADO  1022027   

   OBSERVACAO                                          RESULTADO     GRUPO  \
0         NaN  PESQUISA DE BAAR NEGATIVA EM 100 CAMPOS EXAMIN...  PULMONAR   

  CATEGORIA  
0   ESCARRO  

[1 rows x 21 columns]

In [12]:
tb_grupo_cat_df.query("MATERIAL == 'FRAGMENTO OSSEO'")

Unnamed: 0  Unnamed: 0.1     ID DT_PEDIDO DT_COLETA  ATENDIMENTO  \
305           305           327    327  01/12/14  01/12/14      1576888   
373           373           395    395  02/03/15  02/03/15      1617308   
374           374           396    396  02/03/15  02/03/15      1617308   
997           997          1050   1053  04/01/17  04/01/17      1988768   
1491         1491          1568   1572  05/05/14  05/05/14      1457038   
1492         1492          1569   1573  05/05/14  05/05/14      1457038   
2767         2767          2915   2920  08/12/14  08/12/14      1582159   
2768         2768          2916   2921  08/12/14  08/12/14      1582159   
3234         3234          3401   3406  09/12/16  10/12/16      1959191   
3235         3235          3402   3407  09/12/16  10/12/16      1959191   
3703         3703          3906   3912  11/04/14  14/04/14      1415397   
3704         3704          3907   3913  11/04/14  11/04/14      1415397   
4053         4053          4276   4282  12/01/17  12/01/17      1959191   
4054         4054          4277   4283  12/01/17  12/01/17      1959191   
4545         4545          4796   4803  13/04/15  13/04/15      1636981   
4546         4546          4797   4804  13/04/15  13/04/15      1636981   
7236         7236          7622   7631  20/05/15  20/05/15      1662517   
7237         7237          7623   7632  20/05/15  20/05/15      1662517   
7328         7328          7727   7736  20/08/20  20/08/20      2685037   
7329         7329          7728   7737  20/08/20  20/08/20      2685037   
7330         7330          7729   7738  20/08/20  20/08/20      2685037   
8081         8081          8530   8539  22/11/18  22/11/18      2364274   
8082         8082          8531   8540  22/11/18  22/11/18      2364274   
8083         8083          8532   8541  22/11/18  22/11/18      2364274   
8084         8084          8533   8542  22/11/18  22/11/18      2364274   
8242         8242          8695   8704  23/06/16  23/06/16      1881417   
8243         8243          8696   8705  23/06/16  23/06/16      1881417   
9501         9501         10003  10012  27/01/15  27/01/15      1592173   
9502         9502         10004  10013  27/01/15  27/01/15      1592173   
10261       10261         10813  10822  29/01/18  29/01/18      2202190   
10262       10262         10814  10823  29/01/18  29/01/18      2202190   

       CD_ORI_ATEND      ORIGEM  PRONTUARIO  CD_SETOR  ... CD_CONVENIO  \
305              14  INTERNACAO         158        29  ...           1   
373              14  INTERNACAO         196        28  ...           1   
374              14  INTERNACAO         196        28  ...           1   
997              14  INTERNACAO         483       118  ...           1   
1491             14  INTERNACAO         696       100  ...           1   
1492             14  INTERNACAO         696       100  ...           1   
2767             14  INTERNACAO         196       118  ...           1   
2768             14  INTERNACAO         196       118  ...           1   
3234             14  INTERNACAO        1366       118  ...           1   
3235             14  INTERNACAO        1366       118  ...           1   
3703             14  INTERNACAO        1541        35  ...           1   
3704             14  INTERNACAO        1541        35  ...           1   
4053             14  INTERNACAO        1366       118  ...           1   
4054             14  INTERNACAO        1366       118  ...           1   
4545             14  INTERNACAO        1529        29  ...           1   
4546             14  INTERNACAO        1529        29  ...           1   
7236             14  INTERNACAO        2624        29  ...           1   
7237             14  INTERNACAO        2624        29  ...           1   
7328             14  INTERNACAO        1119        29  ...           1   
7329             14  INTERNACAO        1119        29  ...           1   
7330             14  INTERNACAO        1119        2

In [13]:
print("Quantidade de categorias: ", tb_grupo_cat_df['CATEGORIA'].value_counts().size)
tb_grupo_cat_df['CATEGORIA'].value_counts()

Quantidade de categorias:  16


BRONCOSCOPIA                         4681
ESCARRO                              3307
LÍQUIDO PLEURAL                      1538
LIQUOR                                443
BIÓPSIA                               440
ACOMETIMENTO LINFÁTICO                196
LÍQUIDO EM CAVIDADE ABDOMINAL          97
URINA                                  67
ESCARRO INDUZIDO (BAAR)                66
ACOMETIMENTO MÚSCULOESQUELÉTICO        50
LIQUIDO PERICÁRDICO                    49
ACOMETIMENTO HEMATOLÓGICO              44
FRAGMENTO DE PLEURA                    41
ACOMETIMENTO ÓSSEO                     41
ACOMETIMENTO DE PELE E SUBCUTÂNEO      34
EMPIEMA PLEURAL                        22
Name: CATEGORIA, dtype: int64